phase2/addon/lib/utils.js

```diff
@@ -416,6 +416,25 @@ exports.overridePrefs = function(fileName){
   }
 }
 
+exports.findButtonPlacement = function(name){
+
+  try{
+    let placements = JSON.parse(require("sdk/preferences/service")
+                     .get("browser.uiCustomization.state", null))
+                     .placements;
+
+    for (let k in placements){
+      if (~placements[k].indexOf(name))
+        return k;
+    }
+  }
+  catch(e){
+    return null;
```

Comment:
failing to retrieve the pref is as though the button cannot be find in the whitelisted panels